org.eclipse.lsp4j.jsonrpc/src/test/java/org/eclipse/lsp4j/jsonrpc/test/json/MessageJsonHandlerTest.java

```diff
@@ -780,4 +783,61 @@ public void testNotification_AllOrders() {
 			Assert.assertEquals("dummy://mymodel.mydsl", ((Location)params).uri);
 		});
 	}
+	
+
+	@Test
+	public void testParamUnwrap_JsonRpc2_0() {
+		MessageJsonHandler handler = createSimpleRequestHandler(String.class, Boolean.class);
+
+		var request = new RequestMessage();
+		request.setId(1);
+		request.setMethod(handler.getMethodProvider().resolveMethod(null));
+		request.setParams(new boolean[] { true }); // fake wrapped primitive [true] for JsonRpc 2.0
+
+		RequestMessage message = (RequestMessage) handler.parseMessage(request.toString());
+
+		// Check parse - unwrap primitive
+		assertEquals(true, message.getParams());
+
+	}
+
+	@Test
+	public void testArrayParamUnwrap_JsonRpc2_0() {
+		MessageJsonHandler handler = createSimpleRequestHandler(String.class, new TypeToken<List<Boolean>>() {
+		}.getType());
+
+		var request = new RequestMessage();
+		request.setId(1);
+		request.setMethod(handler.getMethodProvider().resolveMethod(null));
+		request.setParams(List.of(new Boolean[] { true })); // fake wrapped array [[true]] for JsonRpc 2.0
+
+		RequestMessage message = (RequestMessage) handler.parseMessage(request.toString());
+
+		// Check parse - unwrap array
+		assertEquals(request.getParams(), message.getParams());
+
+	}
```

Comment:
I don't think this test tests what you intended. It is testing `[true]` not `[[true]]` because the `List.of()` overload that is chosen is `List.of(E... elements)`

FWIW I stared at this for a while before I understood what is going on here and realized the unexpected overload was chosen. Perhaps it would be better to use the same style as the other tests and make the JSON explicit? 

```suggestion
	@Test
	public void testArrayParamUnwrap_JsonRpc2_0() {
		MessageJsonHandler handler = createSimpleRequestHandler(String.class, new TypeToken<List<Boolean>>() {
		}.getType());

		var request = "{\n"
				+ "  \"jsonrpc\": \"2.0\",\n"
				+ "  \"id\": 1,\n"
				+ "  \"method\": \"testMethod\",\n"
				+ "  \"params\": [\n"
				+ "    [\n"
				+ "      true\n"
				+ "    ]\n"
				+ "  ]\n"
				+ "}";
		RequestMessage message = (RequestMessage) handler.parseMessage(request);

		// Check parse - unwrap array
		assertEquals(List.of(true), message.getParams());

	}
```